client/src/js/app.js

```diff
@@ -680,6 +680,103 @@ elmApp.ports.askFromIndexDb.subscribe(function(info) {
       })();
       break;
 
+    case 'IndexDbQueryUploadScreenshot':
+      (async () => {
+
+        let result = await dbSync
+            .whatsAppUploads
+            .where('syncStage')
+            // On stage 0, we upload the file to backend.
+            .equals(0)
+            // We upload screenshots one by one.
+            .limit(1)
+            .toArray();
+
+        if (!result[0]) {
+          // No screenshots to upload.
+          return sendIndexedDbFetchResult(queryType, {tag: 'Success', result: null});
+        }
+
+        const screenshotsUploadCache = "screenshots-upload";
+        const cache = await caches.open(screenshotsUploadCache);
+
+        result.forEach(async function(row, index) {
+            const cachedResponse = await cache.match(row.screenshot);
+
+            if (cachedResponse) {
+              const blob = await cachedResponse.blob();
+              const formData = new FormData();
+              const imageName = 'whatsapp-upload-' + getRandom8Digits() + '.png';
```

Comment:
one note about naming that it's not random, but it's basically the current time in some format, right? Do we need to ensure that filenames need to be unique or not at this point?

Reply:
(I guess Drupal will do its standard file processing, so not)